lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp

```diff
@@ -445,6 +456,42 @@ void InitLtcModuleBindings(py::module m) {
           };
           return GetTensorsDump(tensors, coverter);
         });
+  m.def("_get_ltc_tensors_ts_device_data_node",
+        [](const std::vector<at::Tensor>& tensors) {
+          std::vector<Node*> roots;
+          for (auto& tensor : tensors) {
+            auto xtensor = TryGetLtcTensor(tensor);
+            roots.push_back(xtensor.GetIrValue().node.get());
+          }
+          auto post_order = Util::ComputePostOrder(roots);
+          std::vector<int64_t> tensor_ids;
+          for (auto nodeptr : post_order) {
+            if (nodeptr->op() == *torch::lazy::ltc_device_data) {
+              const auto* device_data_node = torch::lazy::NodeCast<torch::lazy::DeviceData>(nodeptr, *torch::lazy::ltc_device_data);
+              auto infoptr = device_data_node->data()->info();
+              auto deviceDataInfoPtr = (torch::lazy::LazyGraphExecutor::DeviceDataInfo*) infoptr;
+              tensor_ids.push_back(deviceDataInfoPtr->tensor_id);
```

Comment:
ok, so this `tensor_id` field looks like it gets set from the 'unique_id' field of the LazyTensor::Data.

I _think_ (good to confirm this) that there is a fixed/immutable relationship between
at::Tensor <> LazyTensor <> LazyTensor::DeviceData <> unique_id

in other words i think this id is a never-changing id for a given at::Tensor that is a lazy tensor.  Do you agree?  I am just checking my understanding at this point

Reply:
i'm a little hazy on how this id is used inside lazy_graph_executor for caches.

i think in the new design we want to decouple backend data_ptr from LazyTensor.  We probably want to keep a tight binding between LazyTensor <> Data <> unique_id.

Reply:
> ok, so this `tensor_id` field looks like it gets set from the 'unique_id' field of the LazyTensor::Data.
> 
> I _think_ (good to confirm this) that there is a fixed/immutable relationship between at::Tensor <> LazyTensor <> LazyTensor::DeviceData <> unique_id
> 
> in other words i think this id is a never-changing id for a given at::Tensor that is a lazy tensor. Do you agree? I am just checking my understanding at this point

Yep, I agree. My tests on ModuleSub can (partially) confirms this.

Also here is the code I found that setup the relationship of unique_id between at::Tensor and DeviceData: https://github.com/pytorch/pytorch/blob/lazy_tensor_staging/torch/csrc/lazy/core/tensor.cpp#L423-L427